React/Naiming Conventions.md

```diff
@@ -0,0 +1,8 @@
+## Naming conventions
+
+- JSX/TSX files are named `PascalCase.tsx`
+- JS/TS files are named `snake-case.ts`
+- Components/Containers are imported as `PascalCase`
+- Utils/Singletons are imported as `camelCase`
```

Comment:
Where we use `Singleton` pattern in React?
I would avoid this term because it could be confusing to newbies.
And singleton should be exported as `PascalCalse`.
Example of a singleton:
```
const Singleton = (function () {
    let instance;
 
    function createInstance() {
        return new Object("I am the instance");
    }
 
    return {
        getInstance: function () {
            if (!instance) {
                instance = createInstance();
            }
            return instance;
        }
    };
})();
 
function run() {
    var instance1 = Singleton.getInstance();
    var instance2 = Singleton.getInstance();
 
    alert("Same instance? " + (instance1 === instance2));  
}
```

Reply:
`Singleton` is a pattern. You were probably thinking about global constants like:
```
export const isServer = typeof window === 'undefined';
```